internal/controllers/certmanager.go

```diff
@@ -122,6 +122,33 @@ func (r *Reconciler) setupTLS(ctx context.Context, cr *model.CryostatInstance) (
 		return nil, err
 	}
 
+	// Copy Cryostat CA secret in each target namespace
+	for _, ns := range cr.TargetNamespaces {
+		secret, err := r.GetCertificateSecret(ctx, caCert)
+		if err != nil {
+			return nil, err
+		}
+		namespaceSecret := secret.DeepCopy()
+		namespaceSecret.Namespace = ns
```

Comment:
Hi @mwangggg, this looks good, but you need to tell the Kubernetes client to create this new secret. This part actually sends the request to the Kubernetes API server to [create the secret](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/#create-create-a-secret).

While you can use the [controller-runtime client](https://sdk.operatorframework.io/docs/building-operators/golang/references/client/) directly, I like to use the [`controllerutil.CreateOrUpdate`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate) function instead for things like this. That will create the object if it doesn't exist, or allow you to update it if it does exist. `CreateOrUpdate` takes a callback argument where you set properties that you want to be reflected in the resulting object. In this case, it would just be the secret's `Data` field.